pkg/handler/data_recorder_pubsub.go

```diff
@@ -0,0 +1,89 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/swagger_gen/models"
+	"google.golang.org/api/option"
+
+	"github.com/sirupsen/logrus"
+)
+
+type pubsubRecorder struct {
+	enabled  bool
+	producer *pubsub.Client
+	topic    *pubsub.Topic
+}
+
+// NewPubsubRecorder creates a new Pubsub recorder
+var NewPubsubRecorder = func() DataRecorder {
+	client, err := pubsub.NewClient(
+		context.Background(),
+		config.Config.RecorderPubsubProjectID,
+		option.WithCredentialsFile(config.Config.RecorderPubsubKeyFile),
+	)
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error getting pubsub client")
+	}
+
+	return &pubsubRecorder{
+		producer: client,
+		topic:    client.Topic(config.Config.RecorderPubsubTopicName),
+		enabled:  config.Config.RecorderEnabled,
+	}
+}
+
+func (p *pubsubRecorder) AsyncRecord(r *models.EvalResult) {
+	if !p.enabled {
+		return
+	}
+
+	pr := &pubsubEvalResult{
+		EvalResult: r,
+	}
+
+	payload, err := pr.Payload()
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error marshaling")
+	}
+
+	messageFrame := pubsubMessageFrame{
+		Payload:   string(payload),
+		Encrypted: false,
+	}
+
+	_, err = messageFrame.encode()
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error marshaling")
+	}
+
+	ctx := context.Background()
+	res := p.topic.Publish(ctx, &pubsub.Message{Data: payload})
```

Comment:
if you are using messageFrame, here the payload should be the return of `messageFrame.encode()`

Reply:
That is what I asked earlier, Data is a `[]byte` but encode returns a formatted json string I can cast it into `[]byte` but then what's the point?

Reply:
The point is the metadata fields (e.g. encrypted, compressed, future additional fields) of the frame should also be included into the Data field.

If we don't put it now, future changes are all breaking changes.